src/pextlib1.0/tracelib.c

```diff
@@ -614,10 +616,48 @@ static void dep_check(int sock, char *path) {
         answer(sock, "#");
     }
 
+    if (-1 == fs_cs) {
```

Comment:
I thought that maybe caching it once would be good enough to not slow down stuff each time a file is checked. You're right, though, there could be random mountpoints. Even worse - what if we're checking a file installed in the MacPorts applications dir, which might be on a separate partition than `$[prefix}`? Seems like there's no way around doing the check each time a file is accessed in the first place - short of caching it per-file. Since caching is generally missing in tracelib, I guess checking every time is the best we can do for now.

Speaking of memory leaks, here's another...